UnitTests/GitCommandsTests/ExternalLinks/ExternalLinksManagerIntegrationTests.cs

```diff
@@ -43,12 +43,11 @@ public void Setup()
         [TearDown]
         public void TearDown()
         {
-            _userRoaming.SettingsCache.Dispose();
-            _repoDistributed.SettingsCache.Dispose();
-            _repoLocal.SettingsCache.Dispose();
+            _userRoaming.SettingsCache?.Dispose();
+            _repoDistributed.SettingsCache?.Dispose();
+            _repoLocal.SettingsCache?.Dispose();
 
-            _testHelper.Dispose();
-            _testHelper = null;
+            _testHelper?.Dispose();
```

Comment:
> I don't believe so. If ctor throws, there is no instance to dispose even though a memory may have been allocated.

Although the ctor has created some object, it could have been destroyed by some method.
`TearDown` and `Dispose` should not make assumptions on the lifetime of member objects.

Reply:
> Although the ctor has created some object, it could have been destroyed by some method.

This would indicate a bug. And we must fix such bugs instead of masking them.

Reply:
Member object do not necessarily need to exist until `Dispose`.

Reply:
I'm afraid I can't agree with this statement.
In the unit test fixture all dependencies are constructed before each test (i.e. `Setup()`) and disposed after each test (i.e. `TearDown()`).
So each of the constructed instances must exist when we reach `TearDown`. If one of them doesn't - it is a bug that must be investigated and fixed.

Reply:
We need to undo the other 3:
```cs
            _userRoaming.SettingsCache?.Dispose();
            _repoDistributed.SettingsCache?.Dispose();
            _repoLocal.SettingsCache?.Dispose();
```